src/file/file.ts

```diff
@@ -136,4 +140,30 @@ export class File {
 
     await removeEntryFromDirectory(connection, extendedInfo.podWallet, pathInfo.path, pathInfo.filename, true)
   }
+
+  /**
+   * Shares file information
+   *
+   * @param podName pod where file is stored
+   * @param fullPath full path of the file
+   */
+  async share(podName: string, fullPath: string): Promise<Reference> {
+    assertActiveAccount(this.accountData)
+    assertFullPathWithName(fullPath)
+    assertPodName(podName)
+
+    const connection = this.accountData.connection
+    const extendedInfo = await getExtendedPodsList(
+      connection.bee,
+      podName,
+      this.accountData.wallet!,
+      connection.options?.downloadOptions,
+    )
+
+    const meta = (await getRawMetadata(connection.bee, fullPath, extendedInfo.podAddress)).metadata
+    assertRawFileMetadata(meta)
+    const data = stringToBytes(JSON.stringify(createFileShareInfo(meta, extendedInfo.podAddress)))
```

Comment:
you pass `extendedInfo.podAddress` at function parameter `userAddress`.
Maybe wrong naming on either side?